src/module.js

```diff
@@ -21,14 +21,17 @@ class QuillPasteSmart extends Clipboard {
 
         let text;
         let html;
+        let file;
 
         if ((!e.clipboardData || !e.clipboardData.getData) &&
            (window.clipboardData && window.clipboardData.getData)) {
             // compatibility with older IE versions
             text = window.clipboardData.getData('Text');
+            file = window.clipboardData.items[0];
         } else {
             text = e.clipboardData.getData('text/plain');
             html = e.clipboardData.getData('text/html');
+            file = e.clipboardData.items[0];
```

Comment:
I the property `items` always there and an array?

Reply:
Maybe it would make sense get it by optional chaining. Like `e.clipboardData?.items?.[0]`

Reply:
optional chaining added